### PR TITLE
fix: changelog path update and black format 

### PR DESCRIPTION
**Pull request by ishandhanani**
# rubi-py PR

- Note - **please make sure the title of your PR follows semantic versioning** 
- Ex. fix/feat/refactor: description 
- For a breaking change, that will bump an entire version, please use !. Ex. refactor!: description 

## Description

Updates to the CI/CD flow for rubi-py

## What was the issue?

The changelog is not updated during the CD action. Code is not autoformatted 

## What were the changes?

1. Changing the path to the changelog file
2. Adding a black format hook

## What type of change was this 

- [x] fix - fixing bugs and adding small changes (X.X.X+1)
- [ ] feat - introducing a new feature (X.X+1.X)
- [ ] breaking - a breaking API change (X+1.X.X)

### CODE REVIEW

**Comment by ishandhanani** [comment]
As per our convo @adamamsmith - I've added in a `pre-commit` hook that will run black on files that are in the staging area. This should be good to go

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# forces black to format code before every commit d
+repos:
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
+        language_version: python3.9
```
Comment (adamamsmith) [inline .pre-commit-config.yaml:7]: Can we not use at least `python3.10` as that is what the repo required from the `pyproject.toml`

File: rubi/pyproject.toml [hunk 2]
```diff
@@ -21,6 +21,8 @@ eth-utils = "2.1.0"
 subgrounds = { version = "1.6.0", extras = ["dash"] }
 pyyaml = "6.0.0"
 python-semantic-release = "7.34.3"
+pre-commit = "^3.3.3"
```
Comment (adamamsmith) [inline rubi/pyproject.toml:24]: Is it possible to make these `dev-dependencies`?

When running a bot in production these dependencies aren't needed.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# forces black to format code before every commit d
+repos:
+  - repo: https://github.com/psf/black
+    rev: 23.3.0
+    hooks:
+      - id: black
```
Comment (adamamsmith) [inline .pre-commit-config.yaml:6]: Also a question, does this format the code or just check the formatting?

I would prefer that this is a check that then forces the user to update on their side as otherwise your code changes when you push it. In very rare edge cases this could mean bugs are introduced due to "formatting" changes.

```yaml
      - uses: psf/black@stable
        with:
          options: "--check --diff --color"
```

for this to work you may need to do something like this prior to the `black` step:

```yaml
      - name: Ensure python3 venv is installed for Black
        run: |
          apt -y install python3-venv
```

**Review by adamamsmith — CHANGES_REQUESTED** [review]
I assume all the other changes are just due to updating the formatting.

File: .pre-commit-config.yaml (hunk 3 above)
Comment (ishandhanani) [reply .pre-commit-config.yaml:6]: Good question. The hook only applies black on files that are in the staging area. My thought was when rubi-py is being worked on, black can automatically be added onto files that are being worked on. If the edge case bugs show up, the tests will fail. But I see how this is reliant on our testing coverage

Correct me if I'm wrong Adam. Add black as a github action -> if a file has not been formatted, fail the test and let the user know that the file must be formatted. 

Am good with either but I think the pre commit hook is a bit cleaner.

**Comment by ishandhanani** [comment]
@adamamsmith - added the action that fails if the code has not been linted. See commit [678c534](https://github.com/RubiconDeFi/rubi-py/pull/54/commits/678c534a5f6097c66c823f1d64d124516d37f715) and this [failed workflow](https://github.com/RubiconDeFi/rubi-py/actions/runs/5511568082)

**Comment by ishandhanani** [comment]
@adamamsmith - after full black formatting on rubi, see this passing commit [c2983c9](https://github.com/RubiconDeFi/rubi-py/pull/54/commits/c2983c917387da6987d6c942c496d58f734b3220) and this passing [workflow](https://github.com/RubiconDeFi/rubi-py/actions/runs/5511635441)

**Review by adamamsmith — APPROVED** [review]

**Comment by adamamsmith** [comment]
LGTM 🚀 🚀 🚀

**Review by denverbaumgartner — APPROVED** [review]
